### PR TITLE
Doc PR: explains the numbering scheme of the PCA 10059

### PR DESCRIPTION
**Pull request by JPFrancoia**
Adding a readme for the PCA10059 board, to explain the numbering scheme of the software Pins.

This is just a documentation PR, to spare some trouble to newcomers. Feel free to close if you think it's not necessary.

### CODE REVIEW

**Comment by dpgeorge** [comment]
Thanks for the contribution!

I think this change is good, but I think would be better if it were in the main `ports/nrf/README.md` file because it could relate to other boards with more than GPIO port.

**Comment by JPFrancoia** [comment]
Sure, shouldn't be a problem, but can this numbering scheme be applied to all boards?

**Comment by glennrub** [comment]
I also believe it could be made generic as @dpgeorge suggest (`ports/nrf/README.md`). There are three nrf52840 boards in the port, and more boards might be added in the future.

File: ports/nrf/boards/pca10059/readme.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Pins mapping
+
+Software Pins 0-31 correspond to physical pins 0.x and software Pins 32-63
```
Comment (glennrub) [inline ports/nrf/boards/pca10059/readme.md:3]: I would suggest to change "... software Pins 32-63" to "... software Pins 32-48" as nrf52840 only provides 48 GPIO's. (The same is the case for the nrf53 device which also have 48 GPIOs).

Comment (JPFrancoia) [reply ports/nrf/boards/pca10059/readme.md]: Done.

Comment (glennrub) [reply ports/nrf/boards/pca10059/readme.md]: I'm so sorry!
Terrible mistake from my side. Of course should would be 32 - **47**, not 48.

Comment (JPFrancoia) [reply ports/nrf/boards/pca10059/readme.md]: No problem. I addressed it

**Comment by glennrub** [comment]
Thanks @JPFrancoia , rebased, squshed and merged in 3dcb551d8981bfb370bfe8f467fcde59dd7a916a.

**Pull request closed without merging**